fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/impl/ReferencesRdfContext.java

```diff
@@ -48,21 +64,53 @@ public ReferencesRdfContext(final FedoraResource resource,
         throws RepositoryException {
         super(resource, idTranslator);
         property2triple = new PropertyToTriple(resource.getNode().getSession(), idTranslator);
-        concat(putStrongReferencePropertiesIntoContext());
-        concat(putWeakReferencePropertiesIntoContext());
+        putReferencesIntoContext(resource.getNode());
     }
 
-    private Iterator<Triple> putWeakReferencePropertiesIntoContext() throws RepositoryException {
-        final Iterator<Property> properties = resource().getNode().getWeakReferences();
-
-        return Iterators.concat(Iterators.transform(properties, property2triple));
+    private void putReferencesIntoContext(final Node node) throws RepositoryException {
+        concat(Iterators.concat(Iterators.transform(
+            Iterators.concat(node.getReferences(), node.getWeakReferences()), property2triple)));
+        concat(Iterators.concat(Iterators.transform(Iterators.concat(Iterators.transform(
+            Iterators.concat(node.getReferences(), node.getWeakReferences()), potentialProxies)), triplesForValue)));
     }
 
-    private Iterator<Triple> putStrongReferencePropertiesIntoContext() throws RepositoryException {
-        final Iterator<Property> properties = resource().getNode().getReferences();
-
-        return Iterators.concat(Iterators.transform(properties, property2triple));
-
-    }
+    /* References from LDP indirect containers are generated dynamically by LdpContainerRdfContext, so they won't
+       show up in getReferences()/getWeakReferences().  Instead, we should check referrers to see if they are
+       members of an IndirectContainer and generate the appropriate inbound references. */
+    private Function<Property, Iterator<Value>> potentialProxies = new Function<Property, Iterator<Value>>() {
+        @Override
+        public Iterator<Value> apply(final Property p) {
+            final Set<Value> values = new HashSet<Value>();
+            try {
+                for ( final PropertyIterator it = p.getParent().getProperties(); it.hasNext(); ) {
```

Comment:
This `for` block seems to be duplicating the capabilities found in the `PropertyValueIterator`.
https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/impl/mappings/PropertyValueIterator.java